### PR TITLE
Fix for Plasma 5.25

### PR DESCRIPTION
**Pull request by Curve**
This code replaces old function calls with their new replacements.

### CODE REVIEW

File: src/Model.cc [hunk 1]
```diff
@@ -135,6 +135,12 @@ void Model::start(AnimationKind kind)
 {
     m_kind = kind;
 
+    if (m_window) {
```
Comment (zzag) [inline src/Model.cc:138]: visibility ref is conceptually a different thing. I suggest to store the visibility ref alongside with the model, i.e.

```
QMap<KWin::EffectWindow*, Model> m_models;
```

will become

```
struct AnimationData
{
    Model model;
    EffectWindowVisibileRef visibilityRef;
};
QMap<EffectWindow *, AnimationData> m_animations;
```

Comment (Curve) [reply src/Model.cc]: Done ^^

File: src/YetAnotherMagicLampEffect.cc [hunk 2]
```diff
@@ -141,64 +135,50 @@ void YetAnotherMagicLampEffect::prePaintScreen(KWin::ScreenPrePaintData& data, s
...
-
 void YetAnotherMagicLampEffect::paintWindow(KWin::EffectWindow* w, int mask, QRegion region, KWin::WindowPaintData& data)
 {
     QRegion clip = region;
 
-    auto modelIt = m_models.constFind(w);
-    if (modelIt != m_models.constEnd()) {
-        if ((*modelIt).needsClip()) {
-            clip = (*modelIt).clipRegion();
+    auto modelIt = m_animations.constFind(w);
```
Comment (zzag) [inline src/YetAnotherMagicLampEffect.cc:155]: nit: s/modelIt/it/g

**Comment by zzag** [comment]
Thanks!

**Pull request merged**